### PR TITLE
Show Ongoing and Past Listings separately in their respective pages

### PR DESCRIPTION
**Pull request by karenfrilya97**
### Changes
- Previously, all merchant listings (ongoing and past listings) are shown in both Ongoing Listings and Past Listings pages. This PR shows ongoing and past listings separately in the correct pages.

#### Ongoing Listings page
image

#### Past Listings page
image

### Linked Issues
Part of #4, #6.

### CODE REVIEW

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import {USER_NOT_SIGNED_IN} from 'constants/errors/sign-in-errors';
+
+import {useEffect, useState} from 'react';
+
+import {getAllListings} from 'api';
+import {useMerchantContext} from 'components/merchant/contexts/MerchantContext';
+import {Listing} from 'interfaces';
+
+const useListingsApi = (listingsType: string) => {
```
Comment (chshda) [inline web/src/components/merchant/listings/hooks/useListingsApi.tsx:25]: Can we use an enum for this type?

Comment (ifhuang) [reply web/src/components/merchant/listings/hooks/useListingsApi.tsx:25]: +1

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          merchantId,
+          listingStatus: 'ongoing',
+        });
+      } else {
+        const listingGroups = await Promise.all(
+          ['successful', 'completed', 'unsuccessful'].map(listingStatus =>
+            getAllListings({merchantId, listingStatus})
+          )
+        );
+        listings = listingGroups.flat(1);
```
Comment (ifhuang) [inline web/src/components/merchant/listings/hooks/useListingsApi.tsx:49]: it looks like default is 1 and you can get rid of it. just call flat()

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx (hunk 1 above)
Comment (hzxcaryn) [reply web/src/components/merchant/listings/hooks/useListingsApi.tsx:25]: Yea could do a `type ListingsType = 'past' | 'ongoing;`

File: web/src/components/merchant/listings/index.tsx [hunk 3]
```diff
@@ -35,35 +31,19 @@ const ListingsContainer = styled.div`
...
-
-      const listings = await getAllListings({
-        merchantId: merchant.id.toString(),
-      });
-      setListings(listings);
-    };
-    fetchListings();
-  }, [getMerchant]);
+  const listingsType = hash === '#past-listings' ? 'Past' : 'Ongoing';
+  const {listings} = useListingsApi(listingsType.toLowerCase());
```
Comment (hzxcaryn) [inline web/src/components/merchant/listings/index.tsx:35]: Prefer
```suggestion
  const listingsType: ListingsType = hash === '#past-listings' ? 'past' : 'ongoing';
  const {listings} = useListingsApi(listingsType);
```
And then for the `header` style in `MerchantPage` set `text-transform: capitalize;`

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx (hunk 1 above)
Comment (hzxcaryn) [inline web/src/components/merchant/listings/hooks/useListingsApi.tsx:25]: Actually why not just `useListings`?

Comment (karenfrilya97) [reply web/src/components/merchant/listings/hooks/useListingsApi.tsx:25]: @chshda @ifhuang @hzxcaryn Okay! I will do the enum-like string literal like Caryn suggested.

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx (hunk 2 above)
Comment (karenfrilya97) [reply web/src/components/merchant/listings/hooks/useListingsApi.tsx:51]: @ifhuang Okay!

File: web/src/components/merchant/listings/index.tsx (hunk 3 above)
Comment (karenfrilya97) [reply web/src/components/merchant/listings/index.tsx]: @hzxcaryn Thanks! I didn't realise you can do that with text-transform.

File: web/src/components/merchant/listings/hooks/useListingsApi.tsx (hunk 1 above)
Comment (karenfrilya97) [reply web/src/components/merchant/listings/hooks/useListingsApi.tsx]: @hzxcaryn Hmm yeah actually that's better.

**Pull request merged**